src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,284 @@
+package wolox.training.controllers;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import wolox.training.exceptions.responses.BookAlreadyOwnException;
+import wolox.training.exceptions.responses.UsernameAlreadyTakenException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@WebMvcTest(UserController.class)
+class UserControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @Test
+    void whenGetAllUsers_thenReturnJsonArray() throws Exception {
+        //Given
+        User userTest = new User(null,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+        List<User> allUsers = Arrays.asList(userTest);
+
+        given(userRepository.findAll()).willReturn(allUsers);
+
+        //When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/users")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",hasSize(1)))
+                .andExpect(jsonPath("$[0].name", is(userTest.getName())))
+                .andExpect(jsonPath("$[0].birthdate", is(userTest.getBirthdate().toString())));
+        System.out.println("whenGetAllUsers Passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console